src/commands/deploy.ts

```diff
@@ -16,140 +13,65 @@ export default class Deploy extends Command {
-            this.error('TypeScript compiler error(s) occurred');
-            this.exit(1);
-            return;
-        }
-
-        const packager = new AppPackager(this, fd);
-        const zipName = await packager.zipItUp();
-
-        cli.action.stop('packaged!');
-
-        if (!flags.url) {
-            flags.url = await cli.prompt('What is the server\'s url (include https)?');
-        }
-
-        if (!flags.username && !flags.token) {
-            flags.username = await cli.prompt('What is the username?');
-        }
-
-        if (!flags.password && !flags.token) {
-            flags.password = await cli.prompt('And, what is the password?', { type: 'hide' });
-        }
-
         if (flags.i2fa) {
             flags.code = await cli.prompt('2FA code', { type: 'hide' });
         }
-
-        cli.action.start(`${ chalk.green('deploying') } your app`);
-
-        const data = new FormData();
-        data.append('app', fs.createReadStream(fd.mergeWithFolder(zipName)));
-
-        await this.asyncSubmitData(data, flags, fd);
-
-        cli.action.stop('deployed!');
-    }
-
-    // tslint:disable:promise-function-async
-    private async asyncSubmitData(data: FormData, flags: { [key: string]: any }, fd: FolderDetails): Promise<void> {
-        let authResult;
-
-        if (!flags.token) {
-            let credentials: { username: string, password: string, code?: string };
-            credentials = { username: flags.username, password: flags.password };
-            if (flags.code) {
-                credentials.code = flags.code;
-            }
-
-            authResult = await fetch(this.normalizeUrl(flags.url, '/api/v1/login'), {
-                method: 'POST',
-                headers: {
-                    'Content-Type': 'application/json',
+        let serverInfo: IServerInfo1 | IServerInfo2;
+        const tasks = new Listr([
+            {
+                title: 'Checking Report',
+                task: () => {
+                    checkReport(this, fd, flags);
+                    return;
                 },
-                body: JSON.stringify(credentials),
-            }).then((res: Response) => res.json());
-
-            if (authResult.status === 'error' || !authResult.data) {
-                throw new Error('Invalid username and password or missing 2FA code (if active)');
-            }
-        } else {
-            const verificationResult = await fetch(this.normalizeUrl(flags.url, '/api/v1/me'), {
-                method: 'GET',
-                headers: {
-                    'Content-Type': 'application/json',
-                    'X-Auth-Token': flags.token,
-                    'X-User-Id': flags.userid,
+            },
+            {
+                title: 'Getting Server Info',
+                task: async ()  => {
+                    try {
+                        serverInfo = await getServerInfo(fd);
+                    } catch (e) {
+                        throw new Error(e.message);
```

Comment:
If the `e` is `undefined` then an error will occur here. Please follow the above pattern https://github.com/RocketChat/Rocket.Chat.Apps-cli/pull/64/files#diff-2916a9eae992830600ca3418c2c75814R28 to make your code more robust.